api/bibliotheca.py

```diff
@@ -1136,6 +1136,10 @@ class DateResponseParser(BibliothecaParser):
 
     def process_all(self, string):
         parser = etree.XMLParser()
+        # If the data is an HTTP response, it is a bytestring and
+        # must be converted before it is parsed.
+        if isinstance(string, bytes):
+            string = string.decode("utf-8")
         root = etree.parse(StringIO(string), parser)
```

Comment:
Presumably we could also use BytesIO instead; I copied this code from elsewhere in the file.